### ISSUE TITLE
take_along_axis fails with jax2tf shape polymorphism

### ISSUE THREAD

**Issue by sbodenstein** [google/jax#9552]
This seems like it should work, but fails:
```
import jax
from jax.experimental import jax2tf
from jax import numpy as jnp
import tensorflow as tf

f_jax = lambda x,y: jnp.take_along_axis(x, y, axis=1)

x = jnp.ones((5,2))
inds = jnp.zeros((5,1))

f_tf = jax2tf.convert(f_jax, polymorphic_shapes=[jax2tf.PolyShape('b', 2), jax2tf.PolyShape('b', 1)])
tf.function(f_tf).get_concrete_function(tf.TensorSpec([None, 2], tf.float32), tf.TensorSpec([None,1], tf.float32))
```
with error
```
"/export/hda3/borglet/remote_hdd_fs_dirs/0.colab_kernel_deepmind_tpu_dd_py3_sbodenstein.kernel.sbodenstein.1456309314330.14b334fb3717c109/mount/server/origami_notebook_tpu.par/google3/third_party/py/jax/_src/numpy/lax_numpy.py", line 5627, in <listcomp>
        index_dims = [i for i, idx in enumerate(idx_shape) if i == axis or idx != 1]
    File "/export/hda3/borglet/remote_hdd_fs_dirs/0.colab_kernel_deepmind_tpu_dd_py3_sbodenstein.kernel.sbodenstein.1456309314330.14b334fb3717c109/mount/server/origami_notebook_tpu.par/google3/third_party/py/jax/experimental/jax2tf/shape_poly.py", line 278, in __ne__
        return not self.eq(other)
    File "/export/hda3/borglet/remote_hdd_fs_dirs/0.colab_kernel_deepmind_tpu_dd_py3_sbodenstein.kernel.sbodenstein.1456309314330.14b334fb3717c109/mount/server/origami_notebook_tpu.par/google3/third_party/py/jax/experimental/jax2tf/shape_poly.py", line 273, in eq
        raise InconclusiveDimensionOperation(f"Dimension polynomial comparison '{self}' == '{other}' is inconclusive")

    InconclusiveDimensionOperation: Dimension polynomial comparison 'b' == '1' is inconclusive
```
This only seems to fail when `axis!=0`.

@gnecula
